internal/pkg/status/legacy_status.go

```diff
@@ -0,0 +1,238 @@
+		case "Progressing": //appsv1.DeploymentProgressing:
+			// https://github.com/kubernetes/kubernetes/blob/a3ccea9d8743f2ff82e41b6c2af6dc2c41dc7b10/pkg/controller/deployment/progress.go#L52
+			status := clientu.GetStringField(c, "status", "")
+			reason := clientu.GetStringField(c, "reason", "")
+			if status != "True" || reason != "NewReplicaSetAvailable" {
+				progress = false
+			}
+		case "Available": //appsv1.DeploymentAvailable:
+			status := clientu.GetStringField(c, "status", "")
+			if status == "False" {
+				available = false
+			}
+		}
+	}
+
+	return progress && available, nil
+}
+
+// Replicaset
+func replicasetReady(u *unstructured.Unstructured) (bool, error) {
+	failure := false
+	conditions := clientu.GetConditions(u.UnstructuredContent())
+
+	for _, c := range conditions {
+		switch clientu.GetStringField(c, "type", "") {
+		// https://github.com/kubernetes/kubernetes/blob/a3ccea9d8743f2ff82e41b6c2af6dc2c41dc7b10/pkg/controller/replicaset/replica_set_utils.go
+		case "ReplicaFailure": //appsv1.ReplicaSetReplicaFailure
+			status := clientu.GetStringField(c, "status", "")
+			if status == "True" {
+				failure = true
+				break
+			}
+		}
+	}
+
+	c1, err := compareIntFields(u, []string{"status", "replicas"}, []string{"status", "readyReplicas"}, equalInt)
+	if err != nil {
+		return c1, err
+	}
+	c2, err := compareIntFields(u, []string{"status", "replicas"}, []string{"status", "availableReplicas"}, equalInt)
+	if err != nil {
+		return c2, err
+	}
+
+	return !failure && c1 && c2, nil
+}
+
+// Daemonset
+func daemonsetReady(u *unstructured.Unstructured) (bool, error) {
+	c1, err := compareIntFields(u, []string{"status", "desiredNumberScheduled"}, []string{"status", "numberAvailable"}, equalInt)
+	if err != nil {
+		return c1, err
+	}
+	c2, err := compareIntFields(u, []string{"status", "desiredNumberScheduled"}, []string{"status", "numberReady"}, equalInt)
+	if err != nil {
+		return c2, err
+	}
+
+	return c1 && c2, nil
+}
+
+// PVC
+func pvcReady(u *unstructured.Unstructured) (bool, error) {
+	val, found, err := unstructured.NestedString(u.UnstructuredContent(), "status", "phase")
+	if err != nil {
+		return false, err
+	}
+	if !found {
+		return false, fmt.Errorf(".status.phase not found")
+	}
+	return val == "Bound", nil // corev1.ClaimBound
+}
+
+// Pod
+func podReady(u *unstructured.Unstructured) (bool, error) {
+	conditions := clientu.GetConditions(u.UnstructuredContent())
+
+	for _, c := range conditions {
+		if clientu.GetStringField(c, "type", "") == "Ready" && (clientu.GetStringField(c, "status", "") == "True" || clientu.GetStringField(c, "reason", "") == "PodCompleted") {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// PodDisruptionBudget
+func pdbReady(u *unstructured.Unstructured) (bool, error) {
+	return compareIntFields(u, []string{"status", "currentHealthy"}, []string{"status", "desiredHealthy"}, geInt)
+}
+
+// Cronjob
+func cronjobReady(u *unstructured.Unstructured) (bool, error) {
```

Comment:
I originally took this implementation from Spinnaker: https://github.com/spinnaker/clouddriver/blob/master/clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/handler/KubernetesCronJobHandler.java#L83

However, after playing around with CronJobs, I have found that the status field is never empty, even after immediately checking upon creation:

```
➜  kubectl apply -f test/configs-all/cronjob.yaml -o yaml
apiVersion: batch/v1beta1
kind: CronJob
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"batch/v1beta1","kind":"CronJob","metadata":{"annotations":{},"name":"test-app-cronjob","namespace":"default"},"spec":{"jobTemplate":{"spec":{"template":{"spec":{"containers":[{"args":["echo","hi"],"image":"busybosdx","name":"test-app"}],"restartPolicy":"OnFailure"}}}},"schedule":"*/1 * * * *"}}
  creationTimestamp: 2019-06-10T19:21:41Z
  name: test-app-cronjob
  namespace: default
  resourceVersion: "5210494"
  selfLink: /apis/batch/v1beta1/namespaces/default/cronjobs/test-app-cronjob
  uid: f9c03664-8bb4-11e9-8840-42010a8e00dc
spec:
  concurrencyPolicy: Allow
  failedJobsHistoryLimit: 1
  jobTemplate:
    metadata:
      creationTimestamp: null
    spec:
      template:
        metadata:
          creationTimestamp: null
        spec:
          containers:
          - args:
            - echo
            - hi
            image: busybosdx
            imagePullPolicy: Always
            name: test-app
            resources: {}
            terminationMessagePath: /dev/termination-log
            terminationMessagePolicy: File
          dnsPolicy: ClusterFirst
          restartPolicy: OnFailure
          schedulerName: default-scheduler
          securityContext: {}
          terminationGracePeriodSeconds: 30
  schedule: '*/1 * * * *'
  successfulJobsHistoryLimit: 3
  suspend: false
status: {}
```

It looks like the status != null check is a relic left over from this PR: https://github.com/spinnaker/clouddriver/pull/2691/files#diff-289e93d49fbde6a47d8070e1c2242cc3R89

Unless you hear otherwise, I think it's safe to say that CronJobs are always ready. I am going to remove it from the Checkbox CD doc.